### ISSUE TITLE
Egress controllers not passing pagination constraints

### ISSUE THREAD

**Issue by always-akshat** [sleelin/scimmy-routers#13]
A GET request on `/Users` with pagination constraints does not pass the constraints to the egress handler. 

Here's my setup

`app.ts`
```ts
SCIMMY.Resources.declare(SCIMMY.Resources.User)
    .ingress(UserController.ingress)
    .egress(UserController.egress)
    .degress((resource: any) => {
        console.log(`Request to delete user ${resource}`);
    });
```

`users.controller.ts`
```ts
export async function egress(resource: UserResource) {
    console.dir(resource, {depth: null});
    const {handle} = findUserQuery;
    return handle({filter: mappedFilters});
}
```

The _resource_ is passed as-is to the controller and when I log the values in `resource`, I see an empty constraints object

`Request`
<img width="859" alt="Screenshot 2024-02-20 at 9 36 10 PM" src="https://github.com/scimmyjs/scimmy-routers/assets/3632790/a350b5e4-37b1-4ab0-a00c-c0b4fe763969">

```js
User { constraints: {} }
```

I expect the _resource_ in the controller to contain a _startIndex_ and _count_ keys. This works well for _filter_ and _attributes_.

**Comment by sleelin** [comment]
Hi @always-akshat,
> I expect the _resource_ in the controller to contain a _startIndex_ and _count_ keys. This works well for _filter_ and _attributes_.

Your expectation is correct, and the observed behaviour is indeed unexpected! 

The underlying issue appears to be with how query string parameters are handled by express - as there's nothing to identify the _type_ of query string parameters, they are all parsed into strings. When the string values of the `startIndex` and `count` parameters are passed to the resource at instantiation, they are discarded, as they are expected to be positive integers as defined by the SCIM protocol.

The solution to this issue involves adding a new route handler to the SCIMMYRouters stack to cast the `startIndex` and `count` parameters to integers. Casting should only occur for each parameter on condition that they are defined, their value is a string, and that the string is a valid integer when it has been cast. 

In the interim the new route handler method (or at least a version compatible with TypeScript) will be as follows, and can safely be inserted in the chain where you call app.use, just prior to where you instantiate the SCIMMYRouters class:
```ts
(req, res, next) => {
    const {query} = req as any;
    
    // Cast pagination query parameters from strings to numbers...
    for (let param of ["startIndex", "count"]) {
        // ...but only if they were defined, were strings, and are valid as numbers
        if (!!query[param] && typeof query[param] === "string" && !Number.isNaN(+query[param])) {
            query[param] = +query[param];
        }
    }
    
    next();
}
```

Thanks,
Sam

**Comment by always-akshat** [comment]
This makes sense. @sleelin, I'd be happy to submit a PR for this. Will try to get that in by next week.

**Comment by sleelin** [comment]
Hi @always-akshat,
I've added the new route handler to the SCIMMYRouters stack and published v1.0.1 to NPM including the change, which should resolve the issue.

Thanks,
Sam
